config/http_config.go

```diff
@@ -515,6 +516,17 @@ func WithHost(host string) HTTPClientOption {
 	})
 }
 
+// TLSConfigExtension modifies the given tls config and settings.
+type TLSConfigExtension func(*tls.Config, TLSRoundTripperSettings) (*tls.Config, TLSRoundTripperSettings, error)
```

Comment:
maybe we can use the same naming as the existing `dialContextFunc DialContextFunc` (and group them)

Reply:
group them meaning what?

Reply:
Put them next to each other in the struct def. And move the func + type defs up next to the dial ones. For better discoverability.
(using the same naming pattern will help with that as well)

Reply:
I also think the naming should match, `tlsConfigFunc TLSConfigFunc`.

Reply:
addressed!